src/vos/vos_layout.h

```diff
@@ -249,6 +249,8 @@ enum vos_krec_bf {
 	KREC_BF_EVT			= (1 << 0),
 	/* The key is punched at time kr_latest */
 	KREC_BF_PUNCHED			= (1 << 1),
+	/* The key has been (or will be) removed */
+	KREC_BF_REMOVED			= (1 << 2),
```

Comment:
I wonder if we are shooting ourselves in the foot by not making the log punch changes ahead of these changes

Reply:
I agree with you that if we can change our punch mechanism to log-based in time, that will save our a lot effort for the DTX patch.

Reply:
I can maybe take a crack at it next week (after I get the docs stuff done).   The biggest challenge may be what to use for the log...small order btree, perhaps?

Reply:
I am not sure about how to implement that. But we still have 9 DTX patches to be handled, this is the first one. I am afraid that whether we have enough time to land all of them before our Beijing F2F meeting if we re-work based on the new log-based punch framework... Anyway, I agree that if we re-work the DTX patch against the log-based punch framework, the new patch possibly be more simple/clean than current implementation.